README.md
# iframe-lightbox

Responsive no-jQuery pure JS/CSS Lightbox for iframes, no dependencies, customizable aspect ratio, 5kb unminified source code, with demo

[![npm](https://img.shields.io/npm/v/iframe-lightbox.svg)](https://www.npmjs.com/package/iframe-lightbox)
[![Build Status](https://travis-ci.com/englishextra/iframe-lightbox.svg?branch=master)](https://travis-ci.com/englishextra/iframe-lightbox)
[![Codacy Badge](https://app.codacy.com/project/badge/Grade/369642c14d3344bebe134c76f0f5dde8)](https://www.codacy.com/manual/englishextra/iframe-lightbox/dashboard?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=englishextra/iframe-lightbox&amp;utm_campaign=Badge_Grade)
[![jsdelivr](https://data.jsdelivr.com/v1/package/npm/iframe-lightbox/badge)](https://www.jsdelivr.com/package/npm/iframe-lightbox)

## Demo

[codepen](https://codepen.io/englishextra/full/jmjayV/)
[jsfiddle](https://fiddle.jshell.net/englishextra/8pzy6uhr/show/)
[jsbin](https://output.jsbin.com/saqine)

## Features

* Simple initialization
* Nicely fits YouTube / Vimeo / SoundCloud / Audiomack or other URL via iframe
* Customizable aspect ratio via `data-padding-bottom` attribute
* Iframe content can be scrollable or not (default)
* Debounced launch, default 500ms, custom rate can be set with `rate` property of options object
* Preloading spinner that is unset after onload event succeeds
* Pure CSS Retina Ready UI images, no external ones (prompted by github.com/jasomdotnet, thanks)
* Custom event callbacks

## CDN

### jsDelivr

`https://cdn.jsdelivr.net/gh/englishextra/iframe-lightbox@latest/js/iframe-lightbox.min.js`
`https://cdn.jsdelivr.net/gh/englishextra/iframe-lightbox@latest/css/iframe-lightbox.min.css`

### unpkg

`https://unpkg.com/iframe-lightbox@latest/js/iframe-lightbox.js`
`https://unpkg.com/iframe-lightbox@latest/css/iframe-lightbox.css`

## Install

### npm

`npm install iframe-lightbox`

## Setup

`class` is not required. `iframe-lightbox-link` is used here to select elements. You may use some other method for elements selection.
`data-src` is another method to get the source URL when you do not want the link to lead to some real URL.
`href` is required, and contains URL of your content.
`data-padding-bottom` is optional, and can be used to change default 16/9 Aspect Ratio to the one of yours with the formula: a percentage value of

```txt
height/width*100
```

For instance, HD would be: 9 / 16 * 100 + "%"
So, for YouTube or Vimeo, `data-padding-bottom="56.25%"` would be enough.
For SoundCloud embedded player via iframe, use: `data-padding-bottom="166px"`.
For Audiomack embedded player via iframe, use: `data-padding-bottom="252px"`.
For Scrollable content set `data-scrolling="true"`, or add `scrolling` option property with `true`.
The `data-scrolling` is optional, makes iframe content scrollable or not (default); this can be set with `scrolling` option property.
For those who don't use 3rd-party scripts that interfere with links behaviour and don't force `window.location` they have no need in either `data-touch="true"` or `{touch: true}`.
When you have scripts that interfere, then to keep lightbox working, use `{touch: true}` or `data-touch="true"`.
`data-src` or `href` doesn't matter, but you shouldn't enable this touch override if you have a full screen image in a lighbox link and have no other space to scroll down.

### YouTube

```html
<a
  class="iframe-lightbox-link"
  href="https://www.youtube.com/embed/KK9bwTlAvgo?autoplay=0"
  data-padding-bottom="56.25%">YouTube</a>
```

### Vimeo

```html
<a
  class="iframe-lightbox-link"
  href="https://player.vimeo.com/video/165424115?autoplay=false"
  data-padding-bottom="56.25%">Vimeo</a>
```

### SoundCloud

```html
<a
  class="iframe-lightbox-link"
  href="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/317031598&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;visual=true"
  data-padding-bottom="166px">SoundCloud</a>
```

### Audiomack

```html
<a
  class="iframe-lightbox-link"
  href="https://audiomack.com/embed/song/lax/go-low"
  data-padding-bottom="252px">Audiomack</a>
```

## Scrollable content

```html
<a
  class="iframe-lightbox-link"
  href="https://www.w3.org/"
  data-scrolling="true">Scrollable content</a>
```

## Initialize

```js
[].forEach.call(document.getElementsByClassName("iframe-lightbox-link"), function (el) {
  el.lightbox = new IframeLightbox(el);
});
```

## Tips

SPA / PWA developers don't need to bother: work-related class is added to a link.
That way you avoid multiple assignments to a single element.

## Examples of event handling

```js
(function(root, document) {
  "use strict";
  [].forEach.call(document.getElementsByClassName("iframe-lightbox-link"), function(el) {
    el.lightbox = new IframeLightbox(el, {
      onCreated: function() {
        /* show your preloader */
      },
      onLoaded: function() {
        /* hide your preloader */
      },
      onError: function() {
        /* hide your preloader */
      },
      onClosed: function() {
        /* hide your preloader */
      },
      scrolling: false,
      /* default: false */
      rate: 500 /* default: 500 */,
      touch: false /* default: false - use with care for responsive images in links on vertical mobile screens */
    });
  });
})("undefined" !== typeof window ? window : this, document);
```

## GitHub

Inspired by [squeral/lightbox](https://github.com/squeral/lightbox)
[englishextra/img-lightbox](https://github.com/englishextra/img-lightbox)

## License

Available under [MIT license](https://opensource.org/licenses/MIT).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## 0.2.9 - 2018-12-22

### Changed

- Fixed body scroll disabling on lightbox open https://github.com/englishextra/iframe-lightbox/issues/18

## 0.2.7 - 2018-12-21

### Changed

- Added touchstart surpress to prevent Wordpress takeover https://github.com/englishextra/iframe-lightbox/issues/17

## 0.2.6 - 2018-12-21

### Changed

- Fixed CSS minification https://github.com/englishextra/iframe-lightbox/issues/17

## 0.2.5 - 2018-12-20

### Changed

- Added touch events support https://github.com/englishextra/iframe-lightbox/issues/16

## 0.2.4 - 2018-12-18

### Changed

- Reorginized the file tree of the library

## 0.2.3 - 2018-12-18

### Changed

- Changed `data-src` (which is still supported for compatibility) to `href` as the source for iframe content https://github.com/englishextra/iframe-lightbox/issues/15

## 0.2.2 - 2018-12-16

### Changed

- Added `data-scrolling="true"`, an alternative to existing `scrolling` option property with `true`. https://github.com/englishextra/iframe-lightbox/issues/6

## 0.2.0 - 2018-12-16

### Changed

- Changed z-index 999999 to play well with wp-admin https://github.com/englishextra/iframe-lightbox/issues/14
- Pure CSS Retina Ready UI images, no external ones, removed inlined SVG and base64 images https://github.com/englishextra/iframe-lightbox/issues/13

## 0.1.9 - 2018-12-15

### Changed

- Freeze body scrolling on lightbox open https://github.com/englishextra/iframe-lightbox/issues/12
- Changed z-index 999 https://github.com/englishextra/iframe-lightbox/issues/10
- close and loading with SMIL SVG, and base64 images for compatibility with IE11 and Edge https://github.com/englishextra/iframe-lightbox/issues/13

## 0.1.8 - 2018-12-15

### Changed

- Added Close button https://github.com/englishextra/iframe-lightbox/issues/8
- Closes on ESC https://github.com/englishextra/iframe-lightbox/issues/7
- Added scrolling option (default: false) https://github.com/englishextra/iframe-lightbox/issues/6

## 0.1.7 - 2018-05-02

### Changed

- Added onClosed callback option

## 0.1.6 - 2017-10-25

### Changed

- Updated the forgotten minified file `iframe-lightbox.min.js`

## 0.1.5 - 2017-10-25

### Changed

- The new `onLoaded` event is now an alias for `onIframeLoaded`

## 0.1.4 - 2017-10-11

### Added

- This CHANGELOG file
- [Event Handling Callbacks #3](https://github.com/englishextra/iframe-lightbox/pull/3)

index.js
/*jslint node: true */

/*jslint esversion: 6 */

/*!
 * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 * https://www.electronjs.org/docs/tutorial/security
 */
const electron = require("electron");
const app = electron.app; /* a life cycle module */
const BrowserWindow = electron.BrowserWindow; /* a browser window module */
const path = require("path");
/* a global link */

/* the window will close once the JS object is cleared */
var mainWindow = null;
/* check if all the app’s windows are closed and shut down the app */
app.on("window-all-closed", function () {
/* in OS X stay active until Cmd + Q is pressed */
	if (process.platform !== "darwin") {
		app.quit();
	}
});
/* called when Electron inits and is ready to create a browser window */
app.on("ready", function () {
/* create the window */

/* https://github.com/electron/electron/blob/master/docs/api/browser-window.md */

/* https://electronjs.org/docs/tutorial/security */
	mainWindow = new BrowserWindow({
			webPreferences: {
				contextIsolation: false,
				worldSafeExecuteJavaScript: true,
				nodeIntegration: true
			},
			width: 1080,
			height: 688,
			icon: "favicon.ico",
			title: "img-lightbox Demo Page"
		});
/* load index.html */
	mainWindow.loadURL(path.join("file://", __dirname, "/index.html"));
/* open DevTools. */

/* mainWindow.webContents.openDevTools(); */

/* gets executed when window close event is generated */
	mainWindow.on("closed", function () {
		/* remove the link to the window */
		mainWindow = null;
	});
/* https://stackoverflow.com/questions/48854265/why-do-i-see-an-electron-security-warning-after-updating-my-electron-project-t */
	process.env.ELECTRON_DISABLE_SECURITY_WARNINGS = "1";
});


gulpfile.js
/*global require */

/*!
 * @see {@link https://github.com/mildrenben/surface/blob/master/gulpfile.js}
 * @see {@link https://www.webstoemp.com/blog/gulp-setup/}
 * @see {@link https://gulpjs.com/plugins/blackList.json}
 * @see {@link https://hackernoon.com/how-to-automate-all-the-things-with-gulp-b21a3fc96885}
 * @see {@link https://stackoverflow.com/questions/36897877/gulp-error-the-following-tasks-did-not-complete-did-you-forget-to-signal-async}
 * @see {@link https://zzz.buzz/2016/11/19/gulp-4-0-upgrade-guide/}
 * @see {@link https://blog.khophi.co/migrate-gulp-4-complete-example/}
 * @see {@link https://www.joezimjs.com/javascript/complete-guide-upgrading-gulp-4/}
 * @see {@link https://codeburst.io/switching-to-gulp-4-0-271ae63530c0}
 */

var gulp = require("gulp");
var plumber = require("gulp-plumber");
var sass = require("gulp-sass");
var minifyCss = require("gulp-clean-css");
var cleanCssOptions = {
	level: {
		1: {
			specialComments: 0
		}
	}
};
var uglify = require("gulp-uglify");
var sourcemaps = require("gulp-sourcemaps");
var rename = require("gulp-rename");

var browserSync = require("browser-sync").create();
var reload = browserSync.reload;

/*!
 * @see {@link https://github.com/browserslist/browserslist/issues/386#issuecomment-567709502}
 */
var autoprefixer = require("gulp-autoprefixer");

/*!
 * @see {@link https://github.com/babel/babel/issues/7910}
 */

var babel = require("gulp-babel");
var babelOptions;
babelOptions = {
	"sourceType": "script",
	"presets": ["@babel/env"],
	"plugins": ["@babel/plugin-transform-object-assign",
		"@babel/plugin-transform-arrow-functions",
		"@babel/plugin-transform-async-to-generator"]
};

/*!
 * @see {@link https://github.com/beautify-web/js-beautify}
 * a JSON-formatted file indicated by the --config parameter
 * a .jsbeautifyrc file containing JSON data at any level of the filesystem above $PWD
 * using external config may cause
 * failure to find it
 * if the input/output files reside higher
 * than the config file itself
 */

/* var beautify = require("gulp-jsbeautifier"); */
var beautifyOptions;
beautifyOptions = {
/* "config": ".jsbeautifyrc", */
	"editorconfig": false,
	"indent_size": 4,
	"indent_char": "\t",
	"indent_with_tabs": true,
	"eol": "\n",
	"end_with_newline": true,
	"indent_level": 0,
	"preserve_newlines": true,
	"max_preserve_newlines": 10,
	"html": {
		"indent_inner_html": true,
		"indent_scripts": false,
		"js": {},
		"css": {}
	},
	"css": {
		"newline_between_rules": true
	},
	"js": {
		"space_in_paren": false,
		"space_in_empty_paren": false,
		"jslint_happy": false,
		"space_after_anon_function": true,
		"space_after_named_function": false,
		"brace_style": "collapse",
		"unindent_chained_methods": false,
		"break_chained_methods": true,
		"keep_array_indentation": true,
		"unescape_strings": false,
		"wrap_line_length": 0,
		"e4x": false,
		"comma_first": false,
		"operator_position": "before-newline"
	}
};

/*!
 * @see {@link https://prettier.io/docs/en/options.html}
 * using external config may cause
 * failure to find it
 * if the input/output files reside higher
 * than the config file itself
 */
var prettier = require("gulp-prettier");
var prettierOptions;
prettierOptions = {
/* "config": ".prettierrc", */
	"tabWidth": 4,
	"useTabs": true,
	"endOfLine": "lf",
	"printWidth:": 0
};

var stripDebug = require("gulp-strip-debug");

var eslint = require("gulp-eslint");

/*!
 * @see {@link https://github.com/sasstools/sass-lint/blob/master/docs/sass-lint.yml}
 * @see {@link https://codebeautify.org/yaml-to-json-xml-csv}
 */
var csslint = require("gulp-sass-lint");
var csslintOptions;
var csslintOptions = {
	"options": {
		"merge-default-rules": false,
		"max-warnings": 50
	},
	"files": {
		"ignore": ["scss/_font-face.scss"]
	},
	"rules": {
		"attribute-quotes": 0,
		"border-zero": 0,
		"brace-style": 0,
		"class-name-format": 0,
		"clean-import-paths": 0,
		"empty-args": 0,
		"empty-line-between-blocks": 0,
		"extends-before-declarations": 0,
		"extends-before-mixins": 0,
		"final-newline": 0,
		"force-attribute-nesting": 0,
		"force-element-nesting": 0,
		"force-pseudo-nesting": 0,
		"function-name-format": 0,
		"hex-length": 0,
		"hex-notation": 0,
		"indentation": 0,
		"leading-zero": 0,
		"mixin-name-format": 0,
		"mixins-before-declarations": 0,
		"nesting-depth": 0,
		"no-color-keywords": 0,
		"no-color-literals": 0,
		"no-css-comments": 0,
		"no-debug": 0,
		"no-duplicate-properties": 0,
		"no-empty-rulesets": 0,
		"no-extends": 0,
		"no-ids": 0,
		"no-important": 0,
		"no-invalid-hex": 0,
		"no-mergeable-selectors": 0,
		"no-misspelled-properties": 0,
		"no-qualifying-elements": 0,
		"no-trailing-zero": 0,
		"no-transition-all": 0,
		"no-url-protocols": 0,
		"no-vendor-prefixes": 0,
		"no-warn": 0,
		"one-declaration-per-line": 0,
		"placeholder-in-extend": 0,
		"placeholder-name-format": 0,
		"property-sort-order": 0,
		"pseudo-element": 0,
		"quotes": 0,
		"shorthand-values": 0,
		"single-line-per-selector": 0,
		"space-after-bang": 0,
		"space-after-colon": 0,
		"space-after-comma": 0,
		"space-around-operator": 0,
		"space-before-bang": 0,
		"space-before-brace": 0,
		"space-before-colon": 0,
		"space-between-parens": 0,
		"trailing-semicolon": 0,
		"url-quotes": 0,
		"variable-for-property": 0,
		"variable-name-format": 0,
		"zero-unit": 0
	}
};

var options = {
	libbundle: {
		src: "./src/*.js",
		js: "./js",
		scss: "./scss/*.scss",
		css: "./css"
	},
};

gulp.task("compile-libbundle-css", function () {
	return gulp.src(options.libbundle.scss)
	.pipe(plumber())
	.pipe(sourcemaps.init())
	.pipe(sass({
			errLogToConsole: true
		}))
	.pipe(autoprefixer())
	.pipe(prettier(prettierOptions))
/* .pipe(beautify(beautifyOptions)) */
	.pipe(gulp.dest(options.libbundle.css))
	.pipe(rename(function (path) {
			path.basename += ".min";
		}))
	.pipe(minifyCss(cleanCssOptions))
	.pipe(sourcemaps.write("."))
	.pipe(gulp.dest(options.libbundle.css))
	.pipe(plumber.stop());
});

gulp.task("lint-libbundle-css", function () {
	return gulp.src(options.libbundle.scss)
	.pipe(csslint(csslintOptions))
	.pipe(csslint.format())
	.pipe(csslint.failOnError());
});

gulp.task("compile-libbundle-js", function () {
	return gulp.src(options.libbundle.src)
	.pipe(plumber())
	.pipe(sourcemaps.init())
	.pipe(babel(babelOptions))
	.pipe(prettier(prettierOptions))
/* .pipe(beautify(beautifyOptions)) */
	.pipe(gulp.dest(options.libbundle.js))
	.pipe(rename(function (path) {
			path.basename += ".min";
		}))
	.pipe(stripDebug())
	.pipe(uglify())
	.pipe(sourcemaps.write("."))
	.pipe(gulp.dest(options.libbundle.js))
	.pipe(plumber.stop());
});

gulp.task("lint-libbundle-js", function () {
	return gulp.src(options.libbundle.src)
	.pipe(eslint())
	.pipe(eslint.format())
	.pipe(eslint.failAfterError());
});

/*!
 * @see {@link https://browsersync.io/docs/gulp}
 */
gulp.task("browser-sync", gulp.series(gulp.parallel(
			"lint-libbundle-js",
			"lint-libbundle-css"), function watchChanges() {

		browserSync.init({
			server: "./"
		});

		gulp.watch("./*.html").on("change", reload);
		gulp.watch("./css/*.css").on("change", reload);
		gulp.watch("./scss/*.scss", gulp.parallel("compile-libbundle-css")).on("change", reload);
		gulp.watch("./js/*.js").on("change", reload);
		gulp.watch("./src/*.js", gulp.parallel("compile-libbundle-js")).on("change", reload);
	}));

gulp.task("default", gulp.task("browser-sync"));


src/_iframe-lightbox.js
/*!
 * @see {@link https://github.com/englishextra/iframe-lightbox}
 * modified Simple lightbox effect in pure JS
 * @see {@link https://github.com/squeral/lightbox}
 * @see {@link https://github.com/squeral/lightbox/blob/master/lightbox.js}
 * @params {Object} elem Node element
 * @params {Object} settings object
 * el.lightbox = new IframeLightbox(elem, settings)
 * passes jshint
 */

/*jslint browser: true */

/*jslint node: true */

/*jshint -W014 */
(function (root, document) {
	"use strict";
	var docElem = document.documentElement || "";
	var docBody = document.body || "";
	var containerClass = "iframe-lightbox";
	var iframeLightboxWindowIsBindedClass = "iframe-lightbox-window--is-binded";
	var iframeLightboxOpenClass = "iframe-lightbox--open";
	var iframeLightboxLinkIsBindedClass = "iframe-lightbox-link--is-binded";
	var isLoadedClass = "is-loaded";
	var isOpenedClass = "is-opened";
	var isShowingClass = "is-showing";
	var isMobile = navigator.userAgent.match(/(iPad)|(iPhone)|(iPod)|(Android)|(PlayBook)|(BB10)|(BlackBerry)|(Opera Mini)|(IEMobile)|(webOS)|(MeeGo)/i);
	var isTouch = isMobile !== null || document.createTouch !== undefined || "ontouchstart" in root || "onmsgesturechange" in root || navigator.msMaxTouchPoints;
	var IframeLightbox = function (elem, settings) {
		var options = settings || {};
		this.trigger = elem;
		this.el = document.getElementsByClassName(containerClass)[0] || "";
		this.body = this.el ? this.el.getElementsByClassName("body")[0] : "";
		this.content = this.el ? this.el.getElementsByClassName("content")[0] : "";
		this.src = elem.dataset.src || "";
		this.href = elem.getAttribute("href") || "";
		this.dataPaddingBottom = elem.dataset.paddingBottom || "";
		this.dataScrolling = elem.dataset.scrolling || "";
		this.dataTouch = elem.dataset.touch || "";
		this.rate = options.rate || 500;
		this.scrolling = options.scrolling;
		this.touch = options.touch;
		this.onOpened = options.onOpened;
		this.onIframeLoaded = options.onIframeLoaded;
		this.onLoaded = options.onLoaded;
		this.onCreated = options.onCreated;
		this.onClosed = options.onClosed;
		this.init();
	};
	IframeLightbox.prototype.init = function () {
		var _this = this;
		if (!this.el) {
			this.create();
		}
		var debounce = function (func, wait) {
			var timeout,
			args,
			context,
			timestamp;
			return function () {
				context = this;
				args = [].slice.call(arguments, 0);
				timestamp = new Date();
				var later = function debounced() {
					var last = (new Date()) - timestamp;
					if (last < wait) {
						timeout = setTimeout(later, wait - last);
					} else {
						timeout = null;
						func.apply(context, args);
					}
				};
				if (!timeout) {
					timeout = setTimeout(later, wait);
				}
			};
		};
		var logic = function () {
			_this.open();
		};
		var handleIframeLightboxLink = function (e) {
			e.stopPropagation();
			e.preventDefault();
			debounce(logic, this.rate).call();
		};
		if (!this.trigger.classList.contains(iframeLightboxLinkIsBindedClass)) {
			this.trigger.classList.add(iframeLightboxLinkIsBindedClass);
			this.trigger.addEventListener("click", handleIframeLightboxLink);
			if (isTouch && (_this.touch || _this.dataTouch)) {
				this.trigger.addEventListener("touchstart", handleIframeLightboxLink);
			}
		}
	};
	IframeLightbox.prototype.create = function () {
		var _this = this,
		backdrop = document.createElement("div");
		backdrop.classList.add("backdrop");
		this.el = document.createElement("div");
		this.el.classList.add(containerClass);
		this.el.appendChild(backdrop);
		this.content = document.createElement("div");
		this.content.classList.add("content");
		this.body = document.createElement("div");
		this.body.classList.add("body");
		this.content.appendChild(this.body);
		this.contentHolder = document.createElement("div");
		this.contentHolder.classList.add("content-holder");
		this.contentHolder.appendChild(this.content);
		this.el.appendChild(this.contentHolder);
		this.btnClose = document.createElement("button");
		this.btnClose.classList.add("btn-close");
		this.el.appendChild(this.btnClose);
		docBody.appendChild(this.el);
		backdrop.addEventListener("click", function () {
			_this.close();
		});
		this.btnClose.addEventListener("click", function () {
			_this.close();
		});
		if (!docElem.classList.contains(iframeLightboxWindowIsBindedClass)) {
			docElem.classList.add(iframeLightboxWindowIsBindedClass);
			root.addEventListener("keyup", function (ev) {
				if (27 === (ev.which || ev.keyCode)) {
					_this.close();
				}
			});
		}
		var clearBody = function () {
			if (_this.isOpen()) {
				return;
			}
			_this.el.classList.remove(isShowingClass);
			_this.body.innerHTML = "";
		};
		this.el.addEventListener("transitionend", clearBody, false);
		this.el.addEventListener("webkitTransitionEnd", clearBody, false);
		this.el.addEventListener("mozTransitionEnd", clearBody, false);
		this.el.addEventListener("msTransitionEnd", clearBody, false);
		this.callCallback(this.onCreated, this);
	};
	IframeLightbox.prototype.loadIframe = function () {
		var _this = this;
		this.iframeId = containerClass + Date.now();
		this.iframeSrc = this.src || this.href || "";
		var html = [];
		html.push('<iframe src="' + this.iframeSrc + '" name="' + this.iframeId + '" id="' + this.iframeId + '" onload="this.style.opacity=1;" style="opacity:0;border:none;" webkitallowfullscreen="true" mozallowfullscreen="true" allowfullscreen="true" height="166" frameborder="no"></iframe>');
		html.push('<div class="half-circle-spinner"><div class="circle circle-1"></div><div class="circle circle-2"></div></div>');
		this.body.innerHTML = html.join("");
		(function (iframeId, body) {
			var iframe = document.getElementById(iframeId);
			iframe.onload = function () {
				this.style.opacity = 1;
				body.classList.add(isLoadedClass);
				if (_this.scrolling || _this.dataScrolling) {
					iframe.removeAttribute("scrolling");
					iframe.style.overflow = "scroll";
				} else {
					iframe.setAttribute("scrolling", "no");
					iframe.style.overflow = "hidden";
				}
				_this.callCallback(_this.onIframeLoaded, _this);
				_this.callCallback(_this.onLoaded, _this);
			};
		})(this.iframeId, this.body);
	};
	IframeLightbox.prototype.open = function () {
		this.loadIframe();
		if (this.dataPaddingBottom) {
			this.content.style.paddingBottom = this.dataPaddingBottom;
		} else {
			this.content.removeAttribute("style");
		}
		this.el.classList.add(isShowingClass);
		this.el.classList.add(isOpenedClass);
		docElem.classList.add(iframeLightboxOpenClass);
		docBody.classList.add(iframeLightboxOpenClass);
		this.callCallback(this.onOpened, this);
	};
	IframeLightbox.prototype.close = function () {
		this.el.classList.remove(isOpenedClass);
		this.body.classList.remove(isLoadedClass);
		docElem.classList.remove(iframeLightboxOpenClass);
		docBody.classList.remove(iframeLightboxOpenClass);
		this.callCallback(this.onClosed, this);
	};
	IframeLightbox.prototype.isOpen = function () {
		return this.el.classList.contains(isOpenedClass);
	};
	IframeLightbox.prototype.callCallback = function (func, data) {
		if (typeof func !== "function") {
			return;
		}
		var caller = func.bind(this);
		caller(data);
	};
	root.IframeLightbox = IframeLightbox;
})("undefined" !== typeof window ? window : this, document);


js/iframe-lightbox.js
/*!
 * @see {@link https://github.com/englishextra/iframe-lightbox}
 * modified Simple lightbox effect in pure JS
 * @see {@link https://github.com/squeral/lightbox}
 * @see {@link https://github.com/squeral/lightbox/blob/master/lightbox.js}
 * @params {Object} elem Node element
 * @params {Object} settings object
 * el.lightbox = new IframeLightbox(elem, settings)
 * passes jshint
 */

/*jslint browser: true */

/*jslint node: true */

/*jshint -W014 */
(function(root, document) {
	"use strict";

	var docElem = document.documentElement || "";
	var docBody = document.body || "";
	var containerClass = "iframe-lightbox";
	var iframeLightboxWindowIsBindedClass = "iframe-lightbox-window--is-binded";
	var iframeLightboxOpenClass = "iframe-lightbox--open";
	var iframeLightboxLinkIsBindedClass = "iframe-lightbox-link--is-binded";
	var isLoadedClass = "is-loaded";
	var isOpenedClass = "is-opened";
	var isShowingClass = "is-showing";
	var isMobile = navigator.userAgent.match(
		/(iPad)|(iPhone)|(iPod)|(Android)|(PlayBook)|(BB10)|(BlackBerry)|(Opera Mini)|(IEMobile)|(webOS)|(MeeGo)/i
	);
	var isTouch =
		isMobile !== null ||
		document.createTouch !== undefined ||
		"ontouchstart" in root ||
		"onmsgesturechange" in root ||
		navigator.msMaxTouchPoints;

	var IframeLightbox = function IframeLightbox(elem, settings) {
		var options = settings || {};
		this.trigger = elem;
		this.el = document.getElementsByClassName(containerClass)[0] || "";
		this.body = this.el ? this.el.getElementsByClassName("body")[0] : "";
		this.content = this.el
			? this.el.getElementsByClassName("content")[0]
			: "";
		this.src = elem.dataset.src || "";
		this.href = elem.getAttribute("href") || "";
		this.dataPaddingBottom = elem.dataset.paddingBottom || "";
		this.dataScrolling = elem.dataset.scrolling || "";
		this.dataTouch = elem.dataset.touch || "";
		this.rate = options.rate || 500;
		this.scrolling = options.scrolling;
		this.touch = options.touch;
		this.onOpened = options.onOpened;
		this.onIframeLoaded = options.onIframeLoaded;
		this.onLoaded = options.onLoaded;
		this.onCreated = options.onCreated;
		this.onClosed = options.onClosed;
		this.init();
	};

	IframeLightbox.prototype.init = function () {
		var _this = this;

		if (!this.el) {
			this.create();
		}

		var debounce = function debounce(func, wait) {
			var timeout, args, context, timestamp;
			return function () {
				context = this;
				args = [].slice.call(arguments, 0);
				timestamp = new Date();

				var later = function later() {
					var last = new Date() - timestamp;

					if (last < wait) {
						timeout = setTimeout(later, wait - last);
					} else {
						timeout = null;
						func.apply(context, args);
					}
				};

				if (!timeout) {
					timeout = setTimeout(later, wait);
				}
			};
		};

		var logic = function logic() {
			_this.open();
		};

		var handleIframeLightboxLink = function handleIframeLightboxLink(e) {
			e.stopPropagation();
			e.preventDefault();
			debounce(logic, this.rate).call();
		};

		if (!this.trigger.classList.contains(iframeLightboxLinkIsBindedClass)) {
			this.trigger.classList.add(iframeLightboxLinkIsBindedClass);
			this.trigger.addEventListener("click", handleIframeLightboxLink);

			if (isTouch && (_this.touch || _this.dataTouch)) {
				this.trigger.addEventListener(
					"touchstart",
					handleIframeLightboxLink
				);
			}
		}
	};

	IframeLightbox.prototype.create = function () {
		var _this = this,
			backdrop = document.createElement("div");

		backdrop.classList.add("backdrop");
		this.el = document.createElement("div");
		this.el.classList.add(containerClass);
		this.el.appendChild(backdrop);
		this.content = document.createElement("div");
		this.content.classList.add("content");
		this.body = document.createElement("div");
		this.body.classList.add("body");
		this.content.appendChild(this.body);
		this.contentHolder = document.createElement("div");
		this.contentHolder.classList.add("content-holder");
		this.contentHolder.appendChild(this.content);
		this.el.appendChild(this.contentHolder);
		this.btnClose = document.createElement("button");
		this.btnClose.classList.add("btn-close");

		this.el.appendChild(this.btnClose);
		docBody.appendChild(this.el);
		backdrop.addEventListener("click", function () {
			_this.close();
		});
		this.btnClose.addEventListener("click", function () {
			_this.close();
		});

		if (!docElem.classList.contains(iframeLightboxWindowIsBindedClass)) {
			docElem.classList.add(iframeLightboxWindowIsBindedClass);
			root.addEventListener("keyup", function(ev) {
				if (27 === (ev.which || ev.keyCode)) {
					_this.close();
				}
			});
		}

		var clearBody = function clearBody() {
			if (_this.isOpen()) {
				return;
			}

			_this.el.classList.remove(isShowingClass);

			_this.body.innerHTML = "";
		};

		this.el.addEventListener("transitionend", clearBody, false);
		this.el.addEventListener("webkitTransitionEnd", clearBody, false);
		this.el.addEventListener("mozTransitionEnd", clearBody, false);
		this.el.addEventListener("msTransitionEnd", clearBody, false);
		this.callCallback(this.onCreated, this);
	};

	IframeLightbox.prototype.loadIframe = function () {
		var _this = this;

		this.iframeId = containerClass + Date.now();
		this.iframeSrc = this.src || this.href || "";
		var html = [];
		html.push(
			'<iframe src="' +
				this.iframeSrc +
				'" name="' +
				this.iframeId +
				'" id="' +
				this.iframeId +
				'" onload="this.style.opacity=1;" style="opacity:0;border:none;" webkitallowfullscreen="true" mozallowfullscreen="true" allowfullscreen="true" height="166" frameborder="no"></iframe>'
		);
		html.push(
			'<div class="half-circle-spinner"><div class="circle circle-1"></div><div class="circle circle-2"></div></div>'
		);
		this.body.innerHTML = html.join("");

		(function(iframeId, body) {
			var iframe = document.getElementById(iframeId);

			iframe.onload = function () {
				this.style.opacity = 1;
				body.classList.add(isLoadedClass);

				if (_this.scrolling || _this.dataScrolling) {
					iframe.removeAttribute("scrolling");
					iframe.style.overflow = "scroll";
				} else {
					iframe.setAttribute("scrolling", "no");
					iframe.style.overflow = "hidden";
				}

				_this.callCallback(_this.onIframeLoaded, _this);

				_this.callCallback(_this.onLoaded, _this);
			};
		})(this.iframeId, this.body);
	};

	IframeLightbox.prototype.open = function () {
		this.loadIframe();

		if (this.dataPaddingBottom) {
			this.content.style.paddingBottom = this.dataPaddingBottom;
		} else {
			this.content.removeAttribute("style");
		}

		this.el.classList.add(isShowingClass);
		this.el.classList.add(isOpenedClass);
		docElem.classList.add(iframeLightboxOpenClass);
		docBody.classList.add(iframeLightboxOpenClass);
		this.callCallback(this.onOpened, this);
	};

	IframeLightbox.prototype.close = function () {
		this.el.classList.remove(isOpenedClass);
		this.body.classList.remove(isLoadedClass);
		docElem.classList.remove(iframeLightboxOpenClass);
		docBody.classList.remove(iframeLightboxOpenClass);
		this.callCallback(this.onClosed, this);
	};

	IframeLightbox.prototype.isOpen = function () {
		return this.el.classList.contains(isOpenedClass);
	};

	IframeLightbox.prototype.callCallback = function (func, data) {
		if (typeof func !== "function") {
			return;
		}

		var caller = func.bind(this);
		caller(data);
	};

	root.IframeLightbox = IframeLightbox;
})("undefined" !== typeof window ? window : this, document);
